CONTRIBUTING.md

```diff
@@ -0,0 +1,124 @@
+- Not possible to achieve through a code mod.
+
+## Pull Requests
+When making a pull request, please note the following guidelines:
+- **Branch name:** use a descriptive name for your branch, following the format of `action-target-details` where possible.
+- **Description:** explain what your pull request changes and _why_.
+- **Credits:** if your modifications are in response to a request or issue, link to it or otherwise provide a "paper trail".
+- **Testing:** if you are submitting a bugfix, describe the issue thouroughly and provide recreation steps, *especially* if it is not an existing [issue](https://github.com/EverestAPI/Everest/issues).
+- **Response:** when you submit a pull request, you are responsible for making sure any questions and change requests on it are answered in a timely manner.
+
+
+# Making Changes
+These are explanations and guidelines for how to make changes to the Everest source code.
+
+## Adding Content
+Content added *must not* change the functionality of the base game, or change any public facing API.  
+While many mods also make use of non-public code it is not as important to maintain backwards compatibility for it.
+
+New features should only be added to improve or un-hardcode vanilla features, especially for general use in code mods (f.e. [StrawberryRegistry](https://github.com/EverestAPI/Everest/blob/dev/Celeste.Mod.mm/Mod/Registry/StrawberryRegistry.cs), [CustomNPC](https://github.com/EverestAPI/Everest/blob/dev/Celeste.Mod.mm/Mod/Entities/CustomNPC.cs), [Custom Events](https://github.com/EverestAPI/Everest/blob/dev/Celeste.Mod.mm/Patches/EventTrigger.cs)).  
+Gameplay mechanics, entities, and other features not present in the base game are usually better suited for a mod.
+
+## Patching
+As with in adding content, patches *must not* change the functionality of the base game, or change any public API.
+
+Everest uses [MonoMod](https://github.com/MonoMod/MonoMod) to patch Celeste, which allows for classes and members to be modified.
+For an explanation of how the MonoMod patcher works, see the [MonoMod GitHub](https://github.com/MonoMod/MonoMod#faq)
+
+<!-- For an in-depth explanation of applying patches, along with best practices and common pitfalls, see the [Patching Guide](). -->
+
+There are two ways to modify Vanilla code, depending on the scope of what is being modified:
+
+### `patch_` classes
+Adding a class with the same name as an existing one, and prefixing it with `patch_` will direct MonoMod to apply it as a patch onto the original class.
+
+By default, any methods that are patched are preserved by MonoMod with an `orig_` prefix, and can be referenced in the patched method:
+```cs
+public extern void orig_DoThing(int val);
+public void DoThing(int val) {
+    int newVal = Transform(val);
+    orig_DoThing(newVal);
+}
+```
+
+There are a few attributes that can be applied to members of the class with different effects:
+- `[MonoModIgnore]` ignore this member, do not patch it.
```

Comment:
Custom attributes on `MonoModIgnore`d members still get handled appropriately though, making it necessary sometimes for MonoModRules. Maybe worth mentioning here or under the MonoModRules section? 😅